modules/internal/DefaultRectangular.chpl

```diff
@@ -1926,12 +1926,13 @@ module DefaultRectangular {
           yield info.theDataChunk(0)(i);
         }
       } else {
-        const stride = viewDom.ranges(1).stride: viewDom.idxType,
-              start  = viewDom.ranges(1).first,
+        const viewDomDim = viewDom.dsiDim(1),
+              stride = viewDomDim.stride: viewDom.idxType,
+              start  = viewDomDim.first,
               first  = info.getDataIndex(start),
               second = info.getDataIndex(start + stride),
               step   = (second-first):chpl__signedType(viewDom.idxType),
-              last   = first + (viewDom.ranges(1).length-1) * step:viewDom.idxType;
+              last   = first + (viewDomDim.length-1) * step:viewDom.idxType;
```

Comment:
Looks fine to me. I think I had been relying on 'viewDom' being a DefaultRectangularDom, so ``.ranges`` would be a valid member.